src/_pytest/python.py

```diff
@@ -143,9 +144,29 @@ def async_fail(nodeid: str) -> None:
     fail(msg, pytrace=False)
 
 
+def async_default_arg_warn(nodeid: str, function_name, param) -> None:
+    msg = (
+        "Test function '"
+        + function_name
+        + "' has a default argument '"
+        + param.name
+        + "="
+        + str(param.default)
+        + "'.\n"
+    )
+    warnings.simplefilter("always", PytestDefaultArgumentWarning)
```

Comment:
Is adding a preexisting filter again  a noop? Else this is a problem

Reply:
Even if it is, it seems odd for pytest to programmatically override the warning filters on its own. What's the reason for this?

Reply:
Previously without this filter, the warnings would include a trackback to the line that made the warning. This is not the desired behavior in my opinion because I intended for this warning to help the user pinpoint the line number for the function that used the default parameters.